Add more test for croatian and added locale

darewreck54 opened the pull request:
https://github.com/airbnb/polyglot.js/pull/113 based off the discussion here.  Put fix for Croatian into this PR.  Added missing locale.

ljharb commented on index.js:
```diff
@@ -59,14 +62,17 @@ var pluralTypes = {
   arabic: ['ar'],
-  chinese: ['id', 'ja', 'ko', 'lo', 'ms', 'th', 'tr', 'zh'],
-  german: ['fa', 'da', 'de', 'en', 'es', 'fi', 'el', 'he', 'hu', 'it', 'nl', 'no', 'pt', 'sv'],
+  chinese: ['id', 'ja', 'ko', 'ko-KR', 'lo', 'ms', 'th', 'th-TH', 'tr', 'zh'],
```
i don't think we need to add the exact locales here, when they're covered by the language code.

darewreck54 replied:
Isn't it better to be more specific then general.  For two reason:
1) It removes confusion?  For example, when i looked at it, I was confused but with the addition of the 
2) In the case when the locale is actually different then the language code.  For example, not sure if in the script languages will the pural rules change between locales (cyrillic vs latin)

It does no harm to include it, but it may not be true the other way.

darewreck54 replied:
I'll leave it as then if the first point is acceptable?  Let me know if there is any other issue that i need to address before you can accept the PR.

Thanks!!

ljharb replied:
sure, that seems fine. i'll review and merge this tomorrow (it'd be great to split up the commit into two: one adding more specific locales, and another solely fixing croatian - but i can do that for you prior to merging, so you're welcome to let it stand as-is)

darewreck54 replied:
Another interesting thing is that it's a little confusing how it's grouped.  Since it's grouping by different plural forms, but using a specific country as the variable names.  Example you incorporate korean into the Chinese group.  That's a big confusing.

ljharb commented:
ok, so our l10n folks got back to me with:
> When there is a number in front of a noun, the noun can have three different forms. For example, the noun "guest" (Croatian: gost) can have these forms:
>
> ending in 1 / 21 / 31 (except 11) ---- gost 
> ending in 2 / 3 / 4 (except 12, 13, 14) ---- gosta
> other 0 / 5 / 6 ...  ---- gostiju  
>
> Sometimes, it can happen that the forms of nouns are the same for groups "2 / 3 / 4", and "0 / 5 / 6", but those are exceptions. 
>
>
> In the example you mentioned, the noun "vote" should actually be translated as "glas" and the forms should be as follows:
>
> ending in 1 / 21 / 31 (except 11) ---- glas
> ending in 2 / 3 / 4 (except 12, 13, 14)  ---- glasa
> other 0 / 5 / 6  ---- glasova  
>
>
> And yes, the noun "komentar" (English: comment or feedback) in the example you provided should say "13 komentara".
> ending in 1 / 21 / 31 (except 11) ---- komentar
> ending in 2 / 3 / 4 (except 12, 13, 14)  ---- komentara
> other  0 / 5 / 6  ---- komentara

Can we update this PR to include all these test cases, and to incorporate this feedback?

darewreck54 commented:
Thanks for the feedback.  I'm verifying this with my team and i'll get back to you as soon as I hear back.

ljharb commented on index.js:
```diff
@@ -44,12 +44,12 @@ var pluralTypes = {
   german: function (n) { return n !== 1 ? 1 : 0; },
   french: function (n) { return n > 1 ? 1 : 0; },
-  russian: function (n) {
+  russian_coatian: function (n) {
```
this seems like a typo?

darewreck54 replied:
addressed the change.  I'm still waiting back from the translators but i added the test and made the changes to reflex your comments.

darewreck54 commented:
@ljharb  I've verified with my team and you were correct with the feedback.  I believe I've addressed all your comments.  Please let me know if there is anything else I need to do before this can get merged in.

darewreck54 commented:
hi, any idea when you can merge this in? 

Thanks,
Derek
